test/typecheck/main.go

```diff
@@ -0,0 +1,351 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// do a fast type check of kubernetes code, for all platforms.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/crypto/ssh/terminal"
+	// TODO(rmmh): remove this when golang/go#23712 is fixed, and the
+	// fix is the current minimum Go version to build Kubernetes.
+	"k8s.io/kubernetes/test/typecheck/srcimporter"
+	"k8s.io/kubernetes/third_party/forked/golang/go/types"
+)
+
+var (
+	verbose   = flag.Bool("verbose", false, "print more information")
+	cross     = flag.Bool("cross", true, "build for all platforms")
+	platforms = flag.String("platform", "", "comma-separated list of platforms to typecheck")
+	timings   = flag.Bool("time", false, "output times taken for each phase")
+	defuses   = flag.Bool("defuse", false, "output defs/uses")
+	serial    = flag.Bool("serial", false, "don't type check platforms in parallel")
+
+	isTerminal = terminal.IsTerminal(int(os.Stdout.Fd()))
+	logPrefix  = ""
+
+	// When processed in order, windows and darwin are early to make
+	// interesting OS-based errors happen earlier.
+	crossPlatforms = []string{
+		"linux/amd64", "windows/386",
+		"darwin/amd64", "linux/arm",
+		"linux/386", "windows/amd64",
+		"linux/arm64", "linux/ppc64le",
+		"linux/s390x", "darwin/386",
+	}
+)
+
+type analyzer struct {
+	fset      *token.FileSet // positions are relative to fset
+	conf      types.Config
+	ctx       build.Context
+	failed    bool
+	platform  string
+	donePaths map[string]interface{}
+}
+
+func newAnalyzer(platform string) *analyzer {
+	ctx := build.Default
+	platSplit := strings.Split(platform, "/")
+	ctx.GOOS, ctx.GOARCH = platSplit[0], platSplit[1]
+	ctx.CgoEnabled = true
+
+	a := &analyzer{
+		platform:  platform,
+		fset:      token.NewFileSet(),
+		ctx:       ctx,
+		donePaths: make(map[string]interface{}),
+	}
+	a.conf = types.Config{
+		FakeImportC: true,
+		Error:       a.handleError,
+		Sizes:       types.SizesFor("gc", a.ctx.GOARCH),
+	}
+
+	a.conf.Importer = srcimporter.New(
+		&a.ctx, a.fset, make(map[string]*types.Package))
+
+	if *verbose {
+		fmt.Printf("context: %#v\n", ctx)
+	}
+
+	return a
+}
+
+func (a *analyzer) handleError(err error) {
+	if e, ok := err.(types.Error); ok {
+		path := e.Fset.Position(e.Pos).String()
+		ignore := false
+		if strings.Contains(e.Msg, "GetOpenAPIDefinitions") {
+			// TODO(rmmh): figure out why this happens.
+			// cmd/kube-apiserver/app/server.go:392:70
+			// test/integration/framework/master_utils.go:131:84
+			ignore = true
+		}
+		if strings.Contains(path, "k8s.io/apiserver/pkg/server/config_test.go") &&
+			strings.Contains(e.Msg, "declared but not used") {
+			//TODO(rmmh): make a PR to fix this.
+			ignore = true
+		}
+		if a.ctx.GOOS == "windows" {
+			// https://github.com/vmware/govmomi/issues/1028
```

Comment:
It looks like this was fixed.

Reply:
It was fixed upstream, but not vendored in yet. This PR is gnarly enough, I'll update that in the future. :-)